commit 0f30670348bee32ecbe2aa7c1717560a83af2a3e
Author: Arjan van de Ven <arjan@linux.intel.com>
Date:   Thu Jun 2 23:36:32 2016 -0500

    UBUNTU: SAUCE: Clear Linux: fix initcall timestamps
    
    Print more finegrained initcall timings
    
    use the tsc instead of the jiffies clock for initcall_debug
    
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/init/main.c b/init/main.c
index a11d792..e7d99f4 100644
--- a/init/main.c
+++ b/init/main.c
@@ -747,16 +747,16 @@ __setup("initcall_blacklist=", initcall_blacklist);
 
 static int __init_or_module do_one_initcall_debug(initcall_t fn)
 {
-	ktime_t calltime, delta, rettime;
+	unsigned long long calltime, delta, rettime;
 	unsigned long long duration;
 	int ret;
 
-	printk(KERN_DEBUG "calling  %pF @ %i\n", fn, raw_smp_processor_id());
-	calltime = ktime_get();
+	printk(KERN_DEBUG "calling  %pF @ %i\n", fn, task_pid_nr(current));
+	calltime = local_clock();
 	ret = fn();
-	rettime = ktime_get();
-	delta = ktime_sub(rettime, calltime);
-	duration = (unsigned long long) ktime_to_ns(delta) >> 10;
+	rettime = local_clock();
+	delta = rettime - calltime;
+	duration = delta >> 10;
 	printk(KERN_DEBUG "initcall %pF returned %d after %lld usecs\n",
 		 fn, ret, duration);
 
